nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -83,6 +83,7 @@ class BtServiceNode : public BT::SyncActionNode
   // The main override required by a BT service
   BT::NodeStatus tick() override
   {
+    request_ = std::make_shared<typename ServiceT::Request>();
```

Comment:
I think this should be in the BtServiceNode constructor, we don't want to allocate each tick. 

The service request takes a raw pointer so we don't need to reallocate each iteration (https://github.com/ros2/rclcpp/blob/f160a8bc1d2b4f9a9fb0bba23d0b5d0f8c1afaab/rclcpp/include/rclcpp/client.hpp#L312)

Reply:
I'm sorry, that is my bad.

Reply:
No worries, thanks for pointing this out :-)